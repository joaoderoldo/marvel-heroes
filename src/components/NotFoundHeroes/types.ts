export interface NotFoundHeroesProps {
  search: string;
  handleCleanSearch: () => void;
}
