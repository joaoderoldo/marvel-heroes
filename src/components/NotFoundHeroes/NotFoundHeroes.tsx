import { Container } from "@/components";

const NotFoundHeroes = ({ search, handleCleanSearch }) => {
  return (
    <Container className="not-found-heroes">
      <h1>
        Não encotramos um herói com a busca: <span>{search}</span>
      </h1>
      <button onClick={handleCleanSearch}>Voltar</button>
    </Container>
  );
};

export default NotFoundHeroes;
