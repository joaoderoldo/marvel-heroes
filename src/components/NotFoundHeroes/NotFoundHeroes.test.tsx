import { vi } from "vitest";
import { render } from "@testing-library/react";
import NotFoundHeroes from "./NotFoundHeroes";

describe("NotFoundHeroes component", () => {
  it("should render correctly with props", () => {
    const props = {
      search: "Spiderman",
      handleCleanSearch: vi.fn(),
    };
    const { container } = render(<NotFoundHeroes {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
