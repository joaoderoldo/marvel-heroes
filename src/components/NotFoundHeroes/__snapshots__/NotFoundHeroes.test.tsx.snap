// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundHeroes component > should render correctly with props 1`] = `
<div
  class="sc-gueYoa gEcUkb not-found-heroes"
>
  <h1>
    Não encotramos um herói com a busca: 
    <span>
      Spiderman
    </span>
  </h1>
  <button>
    Voltar
  </button>
</div>
`;
